src/snapshot/tosnapshot.js

```diff
@@ -0,0 +1,74 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../plotly');
+var Lib = require('../lib');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function toSnapshot(gd, opts) {
```

Comment:
I like that idea, and I think it greatly simplifies programatically creating a snapshot.  Also, I think it will be nice to have a single source for this purpose and the modebar.  I'll work on tidying it up, and then we can figure out best name and where to attach.